components/sup/src/manager.rs

```diff
@@ -961,7 +961,19 @@ impl Manager {
                                       self.feature_flags,
                                       pair.clone());
 
-            self.cleanup_packages(&SUP_PKG_IDENT.parse().unwrap()).await;
+            let fully_qualified_ident =
+                PackageIdent::from_str(&format!("{}/{}", SUP_PKG_IDENT, VERSION)).unwrap();
+            let ident = SUP_PKG_IDENT.parse().unwrap();
+            // Check if we are running the most recent version of the Supervisor. If we are, cleanup
+            // the Supervisor packages. If we are not, it does not make sense to cleanup because we
+            // might uninstall the package that is running.
+            if let Some(package_install) = pkg::installed(&ident) {
```

Comment:
I think we could use an `expect` here (though it might break some tests, or developer workflows, now that I think about it :thinking: )

At the very least, documenting how this could ever be `None` would be good.

Reply:
Right, it would be a rare case, but we could be running a `cargo` built supervisor and have no actual `core/hab-sup` package installed. Addressed in [bb9151](https://github.com/habitat-sh/habitat/pull/7582/commits/bb91519cbbf3e1d4b5a73e2ec01242cc21333e0d)